File: PACE/WebApi/src/PACEWebApiCore.Entities/Administration/RoleAction.cs
using SqlCodeGenerator.Attributes;
using System.ComponentModel.DataAnnotations;

namespace PACEWebApiCore.Entities.Administration
{
   public  class RoleAction:BaseEntity
    {
        [KeyProperty(Identity = true)] public int Id { get; set; }
        [Required] public int RoleId { get; set; }        
        [Required] public int ActionTypeId { get; set; }
        [Required] public bool IsEnabled { get; set; }
    }
}

File: PACE/PaceWebUIApplication/src/app/shared/models/role.action.ts
export class RoleAction {
    public roleId: number;
    public actionTypeId: number;
    private description: string;
    public isEnabled: number;

    constructor(id: number, desc: string, value: number) {
        this.roleId = id;
        this.description = desc;
        this.isEnabled = value;
    }
}
